Delta Core prototype + Delta Standalone extended with Core + Flink extended with Core

tdas opened the pull request.

rahulsmahadev committed:
save

rahulsmahadev committed:
WIP arrow based snapshot

tdas committed:
first cut

tdas committed:
reverted standalone changes

tdas committed:
added core/

tdas committed:
refactored key classes from core to standalone package

tdas committed:
end-to-end working solution

tdas committed:
json and parquet reading working

tdas committed:
standalone with core compiling

scottsand-db commented on core/src/main/java/io/delta/standalone/core/DeltaLogCore.java:
```diff
@@ -0,0 +1,6 @@
+package io.delta.standalone.core;
+
+
+public interface DeltaLogCore {
```
if this is an interface ... then are 'users' (connectors) supposed to instantiate `internal.DeltaLogCoreImpl`?

i.e. standalone provided a `DeltaLog.forTable` static API in its interface .. that could be used to instantiate the internal implementation

tdas committed:
DeltaStandaloneWithCore test passing

scottsand-db committed:
WIP - integrating tasks into splits, creating reader, RowRecord -> RowData, etc.

- copied arrow helpers to a new arrow package
- copied core/test ScanHelper and LogReplayHelper
- created RowRecordToRowData
- created DeltaCoreRowDataReader
- updated DeltaScanTaskCore interface
- updated RowDataFormat (extends DeltaBulkFormat) to return our new DeltaCoreRowDataReader instead

scottsand-db committed:
WIP writing a test

scottsand-db committed:
WIP debugging test

scottsand-db committed:
Fix infinite loop. now have a problem where the ValueRecord isn't serializable

scottsand-db committed:
add comment snippet to show how to create the table for the test

scottsand-db committed:
fixed the serialize bug, still failing for out of range, not sure why

tdas committed:
Flink is working

tdas committed:
refactored to include logscanhelper in standalone instead of being in flink

tdas committed:
Merge pull request #4 from tdas/delta-core-and-flink

Delta Core with Flink working

tdas committed:
Added missing files

tdas committed:
copy test delta tables

tdas committed:
Prepared to add DV support

tdas committed:
added warm up test

tdas committed:
more adjustments
